### ISSUE TITLE
ConnectionImpl leaks in ProtocolStorage

### ISSUE THREAD

**Issue by caoli5288** [ProtocolSupport/ProtocolSupportBungee#63]
![1 jpg](https://user-images.githubusercontent.com/3120587/43701407-ac604be6-9988-11e8-9e80-3ac0db44eb1e.png)
![2 jpg](https://user-images.githubusercontent.com/3120587/43701408-ac938150-9988-11e8-840a-c9d2baeeb418.png)

**Comment by caoli5288** [comment]
Seems can be reproduce by player disconnect by exception in raknet layer

**Comment by Shevchik** [comment]
Does psb connections command shows them?

**Comment by caoli5288** [comment]
@Shevchik 
https://github.com/ProtocolSupport/ProtocolSupportBungee/blob/master/src/protocolsupport/protocol/pipeline/common/LogicHandler.java#L53 That should be connection.getRawAddress()

**Comment by Shevchik** [comment]
Indeed, looks like a forgot to port that change, after implementing other changes back then.
But this will only ever affect your server if you are changing remote ips on bungee level, so you should check if it is really fixed.

**Comment by caoli5288** [comment]
@Shevchik Seems the reason I remove the isLocalAddress check.
Current ip_forward not works so I made this change

**Comment by Shevchik** [comment]
So, did the fix work for you?
